flypipe/node.py

```diff
@@ -274,11 +290,16 @@ def execute(node):
                 ),
             )
 
+            # If cache exists, and the transformation has run, then save its cache
+            runnable_node["node_run_context"].cache_context.write(
+                result.as_type(runnable_node["transformation"].dataframe_type).get_df()

```

Comment:
Hmm is there some edge case you're trying to handle here by setting it as the dataframe type and then pulling out the dataframe? The process_transformation line a few lines above already gets the dataframe and then we're wrapping it in a NodeResult and DataFrameWrapper, you can just capture the result of process_transformation as a separate variable for use here instead of unwrapping the just-wrapped dataframe I think.

Reply:
Commented before explaining selecting only columns defined in the output schema to write the cache

Reply:
Yup makes sense now 👍